test/e2e/framework/fixture.go

```diff
@@ -474,6 +449,64 @@ func (sf SyncerFixture) Start(t *testing.T) *StartedSyncerFixture {
 	// successfully sent a heartbeat to kcp.
 	startedSyncer.WaitForClusterReadyReason(t, ctx, "")
 
+	if useDeployedSyncer {
+		syncerID := syncerConfig.ID()
+
+		// print watch events in syncer namespace. There shouldn't be any.
+		go func() {
+			watch, err := downstreamKubeClient.CoreV1().Events(syncerID).Watch(ctx, metav1.ListOptions{})
+			require.NoError(t, err)
```

Comment:
What's the behavior here since we're in a separate goroutine? This will panic if the test has exited (when `ctx` is cancelled, I imagine)